pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -291,30 +336,30 @@ func (r *Reconciler) reconcile(
 			Dependencies: flow.NewTaskIDs(deployKubeAPIServerService, deployVirtualGardenGardenerAccess, renewVirtualClusterAccess),
 		})
 		rewriteSecretsAddLabel = g.Add(flow.Task{
-			Name: "Labeling secrets to encrypt them with new ETCD encryption key",
+			Name: "Labeling encrypted resources to re-encrypt them with new ETCD encryption key",
 			Fn: flow.TaskFn(func(ctx context.Context) error {
-				return secretsrotation.RewriteEncryptedDataAddLabel(ctx, log, virtualClusterClient, secretsManager, corev1.SchemeGroupVersion.WithKind("SecretList"))
+				return secretsrotation.RewriteEncryptedDataAddLabel(ctx, log, virtualClusterClient, secretsManager, encryptedGVKs...)
 			}).
 				RetryUntilTimeout(30*time.Second, 10*time.Minute).
 				DoIf(helper.GetETCDEncryptionKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationPreparing),
-			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient),
+			Dependencies: flow.NewTaskIDs(initializeVirtualClusterClient, waitUntilGardenerAPIServerReady),
```

Comment:
Shouldn't we wait for gardener-admission-controller, because it handles several secrets like `internal-domain-secret`, `kubeconfig-secret`, `admissionplugin-secret`?

Reply:
Why do you think so? Credentials rotation is typically triggered after a `Garden` was created successfully, and by this time it should be fair to assume that GAC is available. Even if we have updated it before during `deployGardenerAdmissionController` and its pods are currently rolling out, there should always be available replicas. And even if not, labeling the secrets might fail (because the webhook fails) and then the reconciliation is retried anyways, but that's a non-standard case, I'd assume.